### ISSUE TITLE
[BUG] Responses using references are not properly expanded

### ISSUE THREAD

**Issue by dominicbarnes** [OpenAPITools/openapi-generator#16965]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

I've got an API that uses the same response format repeatedly, so I opted for using a `$ref` to avoid the repetition in my spec.

It appears that when you do this, the generator does not seem to correctly pick up the response schema, resulting in it being left out. I've tested this in the `go` and `javascript` generators, so I have a suspicion it's a more widespread issue.

The workaround here is to just avoid `$ref` and repeat the response config each time it is needed.

##### openapi-generator version

7.1.0-SNAPSHOT (latest docker image as of today)

##### OpenAPI declaration file content or url

```yaml
openapi: 3.1.0

info:
  title: required to pass validation
  version: 0.0.0

paths:
  /no_ref:
    get:
      responses:
        '200':
          description: required to pass validation
          content:
            text/plain:
              schema:
                type: string

  /ref/no_ref:
    get:
      responses:
        '200':
          $ref: '#/components/responses/no_ref'

  /ref/ref:
    get:
      responses:
        '200':
          $ref: '#/components/responses/ref'

components:
  responses:
    no_ref:
      description: required to pass validation
      content:
        text/plain:
          schema:
            type: string

    ref:
      description: required to pass validation
      content:
        text/plain:
          schema:
            $ref: '#/components/schemas/simple_text'

  schemas:
    simple_text:
      type: string
```

##### Generation Details

I am doing this on macOS, so I've needed to use docker to work, but I suspect the equivalent CLI instructions should suffice. (there's no extra config to worry about)

```sh
docker run --rm -v "$PWD:/local" openapitools/openapi-generator-cli:latest generate \
	-g go \
	-i /local/spec.yaml \
 	-o /local/client/go

docker run --rm -v "$PWD:/local" openapitools/openapi-generator-cli:latest generate \
	-g javascript \
	-i /local/spec.yaml \
 	-o /local/client/javascript
```

##### Steps to reproduce

Once you have the generated code, here's what I've noticed:

###### go

The `Execute()` methods typically return `(Model, *http.Response, error)`, which is found in `api_default.go`:

```go
// no $ref: works as expected
func (r ApiNoRefGetRequest) Execute() (string, *http.Response, error) {

// uses $ref for response, but no $ref for content: missing model
func (r ApiRefNoRefGetRequest) Execute() (*http.Response, error) {

// uses $ref for response and content: missing model
func (r ApiRefRefGetRequest) Execute() (*http.Response, error) {
```

###### javascript

The `returnType` variables generated typically use the model model or `null` when the response content has no detected schema/model, which is found in `src/api/DefaultApi.js`

```js
// no $ref: works as expected
let returnType = 'String';

// uses $ref for response, but no $ref for content: missing model
let returnType = null;

// uses $ref for response and content: missing model
let returnType = null;
```

##### Related issues/PRs

I have not yet found a ticket that seems to reference this issue as well, as most of my search criteria result in hundreds of issues as they are unfortunately rather common terms.

##### Suggest a fix

n/a

**Comment by wing328** [comment]
> openapi: 3.1.0

A workaround is to use 3.0.1 instead while we're working on better 3.1.0 support for the response schema.

**Comment by wing328** [comment]
merged #16986 to fix the issue.

please give it a try with the latest master
